share/ncs-package/cmake/NcsConfig.cmake

```
@@ -48,11 +48,17 @@ if(NOT NO_BOILERPLATE)
         set(CMAKE_MAKE_PROGRAM ${NCS_TOOLCHAIN_NINJA} CACHE INTERNAL "NCS Toolchain ninja")
       endif()
 
+      if(${CMAKE_HOST_SYSTEM_NAME} STREQUAL Windows)
+        set(env_list_delimeter ";")
```

I've seen a similar solution to this on Stack Overflow, but I'm not able to find it again. There they escaped the `;` with `\\` because it's a special character in cmake lists. I wasn't sure if it was necessary inside strings in a variable, so I tested your changes to be sure (with inverted if-condition). The answer seems to be correct in that the `;` is missing for Windows. Perhaps this has never worked as intended on Windows either?

```suggestion
        set(env_list_delimeter "\\;")
```

For macOS it inserts ":" as expected.

---

> There they escaped the ; with \\ because it's a special character in cmake lists.

Correct, if we were working with this as a CMake list, but we are not.
The `;` is still present in the string, but it is important that when we call `set(ENV{PATH} <val>)` that `<val>` is qouted, to ensure the `;` goes into the environment as a `;`.

>  I wasn't sure if it was necessary inside strings in a variable, so I tested your changes to be sure (with inverted if-condition).

not sure I understand this test ?
You cannot test this inside CMake itself, cause then the variable will be treated as a CMake list.

What you need to do, is to check the actual environment path for any code being invoked from CMake.
For example like this:
```patch
--- a/share/ncs-package/cmake/NcsConfig.cmake
+++ b/share/ncs-package/cmake/NcsConfig.cmake
@@ -66,6 +66,7 @@ if(NOT NO_BOILERPLATE)
       if(NCS_TOOLCHAIN_PYTHONPATH)
         set(ENV{PYTHONPATH} "${NCS_TOOLCHAIN_PYTHONPATH}")
       endif()
+         execute_process(COMMAND c:/tmp/bin/path_test.cmd)
     endif()
   endif()
```
where `path_test.cmd` prints the environment `PATH`, as seen from within that script, like this:
```cmd
REM path_test.cmd
echo OFF

echo "PATH is=%PATH%"
```

This produces the following output:
```
C:\projects\github\ncs\zephyr> echo OFF
"PATH is=C:/opt/ncs-toolchain/v1.8.0/v1.8.0/toolchain/opt/bin;C:/opt/ncs-toolchain/v1.8.0/v1.8.0/toolchain/opt/bin/Scripts;C:/opt/ncs-toolchain/v1.8.0/v1.8.0/toolchain/mingw64/bin;C:\Program Files (x86)\VMware\VMware Player\bin\;C:\Program Files\Git\cmd;C:\Python37\Scripts\;....
```

which is correct.

Now, if I do your proposal, we'll see this instead:
```
C:\projects\github\ncs\zephyr>echo OFF
"PATH is=<shortened>/toolchain/opt/bin;<shortened>/toolchain/opt/bin/Scripts;<shortened>/toolchain/mingw64/bin\;C:\Program Files (x86)\VMware\VMware Player\bin\;
                                                                                        an escaped `;`     ^^^^^^^^
```
which, as you can see, results in an escaped semicolon, like this `\;`.
Now, because we are in windows, and the path delimiter is `\`, then `c:\pathA;c:\pathB`  and `c:\pathA\;c:\pathB` are both valid, so even your suggestion will work.

*note*: `<shortened> == C:/opt/ncs-toolchain/v1.8.0/v1.8.0`

---

Ah, thank you for the explanation. I was testing without quotes in cmake:

    message(FATAL_ERROR $ENV{PATH})

So even though the PATH environment variable had a ";" it would print it without it.